### PR TITLE
feat(api): add simpleActionGuard for flexible resource filtering

### PR DESCRIPTION
**Pull request by Mozan0**

### CODE REVIEW

File: apps/api/src/passport/guard/src/action.guard.ts [hunk 1]
```diff
@@ -349,3 +349,67 @@ function createActionGuard(
...
+          if (statement.resource.include) includedResources.push(...statement.resource.include);
+          if (statement.resource.exclude) excludedResources.push(...statement.resource.exclude);
+        }
+      }
+
+      request.resourceFilter = {include: includedResources, exclude: excludedResources};
+
+      if (matchedStatements.length > 0) {
+        return true;
+      }
```
Comment (htuna07) [inline apps/api/src/passport/guard/src/action.guard.ts]: Could you move this setting resource filter into the if line, otherwise it makes no sense to set the resource filter even if we throw a forbidden exception.
We should also expect the length of includedResources to be greater than zero.
Early return would be better for the error case, by the way.

File: apps/api/src/passport/guard/test/simple.action.guard.spec.ts [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+      });
+
+      expect(await guard()).toBe(true);
+      expect(request.resourceFilter).toEqual({
+        include: [],
+        exclude: ["bucket1/row2"]
+      });
+    });
+  });
+});
```
Comment (htuna07) [inline apps/api/src/passport/guard/test/simple.action.guard.spec.ts:571]: Could you also add edge case test for scenarios like the include is not an array? like string/number etc?

File: apps/api/src/passport/guard/test/simple.action.guard.spec.ts [hunk 3]
```diff
@@ -0,0 +1,472 @@
...
+          }
+        ]
+      });
+
+      expect(await guard()).toBe(true);
+      expect(request.resourceFilter).toEqual({
+        include: ["*/*"],
+        exclude: ["bucket2/*"]
+      });
+    });
```
Comment (htuna07) [inline apps/api/src/passport/guard/test/simple.action.guard.spec.ts:204]: This test is fine, but I'm also wondering what if the **include** array has a single wildcard(*) while the resource i try to access has two params like bucket and row? We should expect the resource filter include array should be single wildcard

**Review by htuna07 — APPROVED** [review]
Seems okay

File: apps/api/src/passport/guard/src/action.guard.ts (hunk 1 above)
Comment (Mozan0) [reply apps/api/src/passport/guard/src/action.guard.ts]: done

File: apps/api/src/passport/guard/test/simple.action.guard.spec.ts (hunk 2 above)
Comment (Mozan0) [reply apps/api/src/passport/guard/test/simple.action.guard.spec.ts:571]: done

File: apps/api/src/passport/guard/test/simple.action.guard.spec.ts (hunk 3 above)
Comment (Mozan0) [reply apps/api/src/passport/guard/test/simple.action.guard.spec.ts:204]: done

**Review by htuna07 — DISMISSED** [review]

**Review by htuna07 — DISMISSED** [review]
